src/asciidocParser.ts

```diff
@@ -5,24 +5,21 @@ import { isNullOrUndefined } from 'util';
 
 const fileUrl = require('file-url');
 
-const asciidoctor = require('@asciidoctor/core')()
+const asciidoctor = require('@asciidoctor/core')
 const docbook = require('@asciidoctor/docbook-converter')
 const kroki = require('asciidoctor-kroki')
 const highlightjsAdapter = require('./highlightjs-adapter')
-highlightjsAdapter.register(asciidoctor)
-
-const use_kroki = vscode.workspace.getConfiguration('asciidoc', null).get('use_kroki');
-if (use_kroki) {
-  kroki.register(asciidoctor.Extensions);
-}
 
 export class AsciidocParser {
     public html: string = '';
     public document = null;
+    public adProcessor = null
```

Comment:
(nitpick): since we are in an `AsciidocParser` we could name it `processor`? as an alternative, I would use the complete name `asciidocProcessor`. `ad` can be confusing if you are not familiar with the code base ("is that a typo, it should be addProcessor no?" 😆)

Reply:
:+1: